cpp/ql/src/semmle/code/cpp/File.qll

```diff
@@ -336,7 +326,13 @@ class File extends Container, @file {
    * for example, for "file.tar.gz", this predicate will have the result
    * "tar.gz", while `getExtension` will have the result "gz".
    */
-  string getExtensions() { files(underlyingElement(this), _, _, result, _) }
+  string getExtensions() {
+    exists(string name, int firstDotPos |
+      name = this.getBaseName() and
+      firstDotPos = min(name.indexOf(".")) and
+      result = name.suffix(firstDotPos + 1)
+    )
```

Comment:
As far as I can tell this change doesn't accurately reproduce current behaviour - despite the comment, for `"file.tar.gz"`, `getExtensions()` would return just `"gz"` and `getShortName()` would be `"file.tar"`.  This has probably been broken at some point, and this is a fix, but we should be aware stuff (like tests) could break.

Reply:
Also, I think the old `getExtensions()` would return an empty string if there's no extension, whereas your code gives no result.  This change is probably worth addressing as it seems like it could potentially break some query somewhere.